es-ds-components/components/es-zip-code-form.vue

```diff
@@ -0,0 +1,204 @@
+<script setup lang="ts">
+const props = defineProps({
+    constrained: {
+        type: Boolean,
+        default: false,
+    },
+    dark: {
+        type: Boolean,
+        default: false,
+    },
+    fieldName: {
+        type: String,
+        default: 'zip_code',
+    },
+    inputId: {
+        type: String,
+        required: true,
+    },
+    newTab: {
+        type: Boolean,
+        default: false,
+    },
+    placeholder: {
+        type: String,
+        default: 'ZIP code',
+    },
+    showPrivacySection: {
+        type: Boolean,
+        default: true,
+    },
+    privacyPolicyLink: {
+        type: String,
+        default: '',
+    },
+    privacyPolicyNewTab: {
+        type: Boolean,
+        default: false,
+    },
+    stackUntil: {
+        type: String,
+        default: '',
+    },
+    url: {
+        type: String,
+        required: true,
+    },
+    zipCodeValue: {
+        type: String,
+        default: '',
+    },
+    selectedProduct: {
+        type: String,
+        default: '',
+    },
+});
+
+const state = reactive({
```

Comment:
nit: I'd recommend using `ref`, mostly for consistency (it should result in the same thing in this context)

Reply:
Consistency with what? I used `state` to match the new[ form validation example](https://new.design.energysage.dev/examples/form-field-validation) as much as possible; 
I do agree that `ref` should probably work the same way for a single value, but when I tried that with `useEsForms` the validation was broken so I'm not sure it's quite supported to not wrap it in `state`.

Reply:
`ref` works differently since the value is accessed through `.value` rather than on the object itself. Vuelidate requires using either `reactive` or `computed` for the form state.

Reply:
(Actually, you can also use an object of refs, like `{ zipCode: ref(props.zipCodeValue) }`. But I think using `reactive` looks cleaner and aligns with the form validation example. [Their docs for reference](https://vuelidate-next.netlify.app/advanced_usage.html#composition-api))

Reply:
gotcha, thanks for the context!